test/common/dns.js

```diff
@@ -0,0 +1,290 @@
+                const txtLength = buffer[offset];
+                rr.entries.push(buffer.toString('utf8',
+                                                position + 1,
+                                                position + 1 + txtLength));
+                position += 1 + txtLength;
+              }
+              assert.strictEqual(position, offset + dataLength);
+              break;
+            }
+          case types.MX:
+            {
+              rr.priority = buffer.readInt16BE(buffer, offset);
+              offset += 2;
+              const { nread, domain } = readDomainFromPacket(buffer, offset);
+              rr.exchange = domain;
+              assert.strictEqual(nread, dataLength);
+              break;
+            }
+          case types.NS:
+          case types.CNAME:
+          case types.PTR:
+            {
+              const { nread, domain } = readDomainFromPacket(buffer, offset);
+              rr.value = domain;
+              assert.strictEqual(nread, dataLength);
+              break;
+            }
+          case types.SOA:
+            {
+              const mname = readDomainFromPacket(buffer, offset);
+              const rname = readDomainFromPacket(buffer, offset + mname.nread);
+              rr.nsname = mname.domain;
+              rr.hostmaster = rname.domain;
+              const trailerOffset = offset + mname.nread + rname.nread;
+              rr.serial = buffer.readUInt32BE(trailerOffset);
+              rr.refresh = buffer.readUInt32BE(trailerOffset + 4);
+              rr.retry = buffer.readUInt32BE(trailerOffset + 8);
+              rr.expire = buffer.readUInt32BE(trailerOffset + 12);
+              rr.minttl = buffer.readUInt32BE(trailerOffset + 16);
+
+              assert.strictEqual(trailerOffset + 20, dataLength);
+              break;
+            }
+          default:
+            throw new Error(`Unknown RR type ${rr.type}`);
+        }
+        offset += dataLength;
+      }
+
+      parsed[sectionName].push(rr);
+
+      assert.ok(offset <= buffer.length);
+    }
+  }
+
+  assert.strictEqual(offset, buffer.length);
+  return parsed;
+}
+
+function writeIPv6(ip) {
+  const parts = ip.replace(/^:|:$/g, '').split(':');
+  const buf = Buffer.alloc(16);
+
+  let offset = 0;
+  for (const part of parts) {
+    if (part === '') {
+      offset += 16 - 2 * (parts.length - 1);
+    } else {
+      buf.writeUInt16BE(parseInt(part, 16), offset);
+      offset += 2;
+    }
+  }
+
+  return buf;
+}
+
+function writeDomainName(domain) {
+  return Buffer.concat(domain.split('.').map((label) => {
+    assert(label.length < 64);
+    return Buffer.concat([
+      Buffer.from([label.length]),
+      Buffer.from(label, 'ascii')
+    ]);
+  }).concat([Buffer.alloc(1)]));
+}
+
+function writeDNSPacket(parsed) {
+  const buffers = [];
+
+  buffers.push(new Uint16Array([
+    parsed.id,
+     // default to “standard response”
+    parsed.flags === undefined ? 0x8180 : parsed.flags,
```

Comment:
·@refack This is not a very helpful comment, I have no idea what you are trying to say.